src/pykeen/trackers/tensorboard.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+
+"""An adapter for TensorBoard."""
+
+import pathlib
+import time
+from typing import Any, Dict, Mapping, Optional, TYPE_CHECKING, Type
+
+from .base import ResultTracker
+from ..constants import PYKEEN_LOGS
+from ..utils import flatten_dictionary
+
+if TYPE_CHECKING:
+    import torch.utils.tensorboard
+
+__all__ = [
+    'TensorBoardResultTracker',
+]
+
+
+class TensorBoardResultTracker(ResultTracker):
+    """A tracker for TensorBoard."""
+
+    #: The class that's used to instantiate a summarywriter
+    summary_writer_cls: Type['torch.utils.tensorboard.SummaryWriter']
+
+    def __init__(
+        self,
+        experiment_path: Optional[str] = None,
+        experiment_name: Optional[str] = None,
+        tags: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        Initialize result tracking via Tensorboard.
+
+        :param experiment_path:
+            The experiment path. A custom path at which the tensorboard logs will be saved.
+        :param experiment_name:
+            The name of the experiment, will be used as a sub directory name for the logging. If no default is given,
+            the current time is used. If set, experiment_path is set, this argument has no effect.
+        :param tags:
+            The additional run details which are presented as tags to be logged
+        """
+        import torch.utils.tensorboard
+        self.summary_writer_cls = torch.utils.tensorboard.SummaryWriter
+        self.tags = tags
+
+        if experiment_path is None:
+            if experiment_name is None:
+                experiment_name = time.strftime('%Y-%m-%d-%H-%M-%S')
+            path = PYKEEN_LOGS.joinpath("tensorboard", experiment_name)
+        elif isinstance(experiment_path, str):
```

Comment:
what happens if neither of these is true? `path` will be unassigned. Should it be assigned the value for `experiment_path`?

Reply:
If the user was passing a `experiment_path` variable that wasn't a string, I wasn't sure how to handle it. I have added a condition in there now to catch that case, but it doesn't seem very elegant. Very open to suggestions!

Reply:
I reorganized this in b9f56110